src/main/java/Duke.java

```diff
@@ -1,10 +1,140 @@
+            String[] commandTypeAndArg = line.split(" ", 2);
+            String commandType = commandTypeAndArg[0];
+            String commandArg = null;
+            if (commandTypeAndArg.length > 1) {
+                commandArg = commandTypeAndArg[1];
+            }
+
+            if (commandType.equals("bye")) {
+                scanner.close();
+                return;
+            }
+            executeCommand(commandType, commandArg);
+        }
+    }
+    
+    private static void executeCommand(String commandType, String commandArg) {
+        switch (commandType) {
+        case "help":
+            printWithBorder(COMMANDS);
+            break;
+        case "list":
+            listAllTasks();
+            break;
+        case "done":
+            markTaskAsDone(commandArg);
+            break;
+        case "todo":
+            addTodo(commandArg);
+            break;
+        case "deadline":
+            addDeadline(commandArg);
+            break;
+        case "event":
+            addEvent(commandArg);
+            break;
+        default:
+            displayInvalidCommandResponse();
+            break;
+        }
+    }
+
+    private static void addTodo(String commandArg) {
+        Todo task = new Todo(commandArg);
+        addTaskToListAndPrintMessage(task);
+    }
+
+    private static void addEvent(String commandArg) {
+        String[] taskDescriptionAndAt = commandArg.split(" /at ", 2);
+        String description = taskDescriptionAndAt[0];
+        String at = taskDescriptionAndAt[1];
+        Event task = new Event(description, at);
+        addTaskToListAndPrintMessage(task);
+    }
+
+    private static void addDeadline(String commandArg) {
+        String[] taskDescriptionAndBy = commandArg.split(" /by ", 2);
+        String description = taskDescriptionAndBy[0];
+        String by = taskDescriptionAndBy[1];
+        Deadline task = new Deadline(description, by);
+        addTaskToListAndPrintMessage(task);
+    }
+
+    private static void addTaskToListAndPrintMessage(Task task) {
+        taskList[taskCount] = task;
+        taskCount += 1;
+        String className = task.getClass().getSimpleName();
+        String taskSuccessfullyAddedMessage = "Alrighty! I have added this new " + className + ":\n    "
+                + task.toString() + "\nYou now have " + Integer.toString(taskCount) + " tasks in the list.";
+        printWithBorder(taskSuccessfullyAddedMessage);
+    }
+
+    private static void displayInvalidCommandResponse() {
+        String invalidCommandResponse = "Invalid command!\n" + COMMANDS;
+        printWithBorder(invalidCommandResponse);
+    }
+
+    private static void markTaskAsDone(String commandArg) {
+        int taskNumber = Integer.parseInt(commandArg);
+        Task task = taskList[taskNumber - 1];
+        task.setIsDone();
+        String taskSuccessfullyMarkedDoneMessage = "Very nice! I've marked this task as done:\n    " 
+                + task.toString();
```

Comment:
Can be merged into a single line to enhance readability

Reply:
thank you! i will merge this into a single line